pennylane/grouping/utils.py

```diff
@@ -0,0 +1,492 @@
+                pauli_vec_1, pauli_vec_2
+            )
+        )
+
+    n_qubits = int(len(pauli_vec_1) / 2)
+
+    for i in range(n_qubits):
+
+        a_x_i = pauli_vec_1[i]
+        a_z_i = pauli_vec_1[n_qubits + i]
+        b_x_i = pauli_vec_2[i]
+        b_z_i = pauli_vec_2[n_qubits + i]
+
+        if not (
+                ((a_x_i == a_z_i == 0) or (b_x_i == b_z_i == 0))
+                or ((a_x_i == b_x_i) and (a_z_i == b_z_i))
+            ):
+            return False
+
+    return True
+
+
+def convert_observables_to_binary(observables):
+    """Converts a list of Pauli words to the binary vector representation and yields column matrix
+        of the binary vectors.
+
+    The dimension of the binary vectors will be implied from the highest wire being acted on
+    non-trivially by the Pauli words in observables.
+
+    **Usage example:**
+
+    >>> convert_observables_to_binary([PauliX(0) @ PauliY(2), PauliZ(0) @ PauliZ(1) @ PauliZ(2)])
+    array([[1., 0.],
+           [0., 0.],
+           [1., 0.],
+           [0., 1.],
+           [0., 1.],
+           [1., 1.]])
+
+    Args:
+        observables (list[Union[Identity, PauliX, PauliY, PauliZ, Tensor]]):
+
+    Returns:
+        array[bool]: a column matrix of the observables in binary vector representation.
+
+    """
+
+    m_cols = len(observables)
+
+    n_qubits = get_n_qubits(observables)
+
+    binary_mat = np.zeros((2 * n_qubits, m_cols))
+
+    for i in range(m_cols):
+        binary_pauli = pauli_to_binary(observables[i], n_qubits)
+
+        binary_mat[:, i] = binary_pauli
+
+    return binary_mat
+
+
+def get_qwc_compliment_adj_matrix(binary_observables):
+    """Obtains the adjacency matrix for the complementary graph of the qubit-wise commutativity
+    graph for a given set of observables in the binary representation.
+
+    The qubit-wise commutativity graph for a set of Pauli words has a vertice for each Pauli word,
+    and two nodes are connected if and only if the corresponding Pauli words are qubit-wise
+    commuting.
+
+    **Usage example:**
+
+    >>> binary_observables
+    array([[1., 0., 0.],
+           [0., 1., 0.],
+           [1., 1., 0.],
+           [0., 1., 1.],
+           [0., 0., 0.],
+           [1., 1., 0.]])
+    >>> get_qwc_compliment_adj_matrix(binary_observables)
+    array([[0., 1., 1.],
+           [1., 0., 0.],
+           [1., 0., 0.]])
+
+    Args:
+        binary_observables (array[bool]): a column matrix of Pauli words in the binary vector
+            representation.
+
+    Returns:
+        array[bool]: the adjacency matrix for the complement of the qubit-wise commutativity graph.
+
+    Raises:
+        ValueError: if input binary observables contain components which are not strictly binary.
+
+    """
+
+    if not np.array_equal(binary_observables, binary_observables.astype(bool)):
+        raise ValueError("Expected a binary array, instead got {}".format(binary_observables))
+
+    if len(np.shape(binary_observables)) == 1:
```

Comment:
Do we need this check?

Reply:
Which check? The if statement on line 473 raises an error if the input matrix is not over the binary field, since the function scope of `get_qwc_compliment_adj_matrix` is to construct the adjacency matrix for the compliment of the QWC graph for a set of Pauli words in the binary representation, input as a matrix. The if statement on line 476 is for catching edge-case where binary observables is a vector, i.e. only 1 Pauli word is specified. The number of Pauli words is inferred from the number of columns in `binary_observables`. If `binary_observables` is a vector, then `np.shape(binary_observables)[1]` will cause an error, since `np.shape(binary_observables)` is of length 1.

Reply:
Fair enough. I'd probably clarify that a single row column is also allowed as input (should it be allowed)? It might be easier to just reshape binary_observables in that case.

Reply:
The binary matrix being a single column is a trivial edge case, since the graph consists of a single node (1x1 adjacency matrix). I'm not sure why you would want to do this, but I suppose it should be possible? I've implemented a reshape of `binary_observables` in the case of having no second element in np.shape.